File: 03 calse lista de enteros/py/lista-enteros.py
class ListaEnteros:
    def __init__(self):
        self.lista = []
        
    def agregar(self, valor):
        self.lista.append(valor)
    
    def eliminar_por_indice(self, indice):
        if 0 <= indice < len(self.lista):
            del self.lista[indice]
            
    def buscar(self, valor):
        try: 
            return self.lista.index(valor)
        except ValueError:
            return -1
    
    def ordenar(self):
        self.lista.sort()     

mi_lista = ListaEnteros()
mi_lista.agregar(5)
mi_lista.agregar(2)
print(mi_lista.lista) #imprime [5, 2]

File: 03 calse lista de enteros/js/lista-enteros.js
class ListaEnteros {
    constructor() {
        this.lista = [];
    }

    agregar(valor) {
        this.lista.push(valor);
    }

    eliminar(valor) {
        const index = this.lista.indexOf(valor);
        if (index !== -1) {
            this.lista.splice(index, 1);
        }
    }

    buscar(valor) {
        return this.lista.indexOf(valor);
    }

    ordenar() {
        this.lista.sort((a, b) => a - b); // Orden ascendente
    }
}

let miLista = new ListaEnteros();

function actualizarResultado() {
    document.getElementById('result').innerHTML = `Resultado: [ ${miLista.lista.join(', ')} ]`;
}

function agregar() {
    const valor = parseInt(document.getElementById('inputValor').value);
    if (!isNaN(valor)) {
        miLista.agregar(valor);
        actualizarResultado();
        document.getElementById('inputValor').value = ''; // Limpiar input
    }
}

function eliminar() {
    const valor = parseInt(document.getElementById('inputValor').value);
    if (!isNaN(valor)) {
        miLista.eliminar(valor);
        actualizarResultado();
        document.getElementById('inputValor').value = ''; // Limpiar input
    }
}

function buscar() {
    const valor = parseInt(document.getElementById('inputValor').value);
    if (!isNaN(valor)) {
        const index = miLista.buscar(valor);
        alert(index !== -1 ? `El número ${valor} está en la posición ${index}` : `${valor} no se encuentra en la lista.`);
    }
}

function ordenar() {
    miLista.ordenar();
    actualizarResultado();
}